src/main/java/com/artipie/asto/fs/FileStorage.java

```diff
@@ -154,44 +156,40 @@ public FileStorage(final Path path, final ExecutorService exec) {
     public CompletableFuture<Void> save(final Key key, final Content content) {
         return CompletableFuture.supplyAsync(
             () -> {
-                final Path file = this.path(key);
-                try {
-                    Files.createDirectories(file.getParent());
-                } catch (final IOException iex) {
-                    throw new UncheckedIOException(iex);
-                }
-                return file;
-            },
-            this.exec
```

Comment:
@olegmoz don't we need to use same executor service for `mkdirs` as for write and move?